src/mpid/common/shm/mpidu_init_shm.c

```diff
@@ -233,6 +199,9 @@ int MPIDU_Init_shm_finalize(void)
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDU_INIT_SHM_FINALIZE);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDU_INIT_SHM_FINALIZE);
 
+#ifdef MPL_USE_LOCK_BASED_PRIMITIEVS
+    MPL_atomic_interprocess_lock_free();
+#endif
```

Comment:
I believe `MPL_atomic_interprocess_lock_free()` should be done after all the possible `Init_shm_barrirer()` since `Init_shm_barrirer()` relies on an interprocess lock (I guess).